prophesee_ros_driver/src/prophesee_ros_publisher.cpp

```diff
@@ -141,22 +191,52 @@ void PropheseeWrapperPublisher::publishCDEvents() {
                     return;
 
                 if (ev_begin < ev_end) {
-                    // Define the message for a buffer of CD events
-                    prophesee_event_msgs::EventArray event_buffer_msg;
+                    // Compute the current local buffer size with new CD events
                     const unsigned int buffer_size = ev_end - ev_begin;
-                    event_buffer_msg.events.resize(buffer_size);
 
-                    // Header Timestamp of the message
-                    event_buffer_msg.header.stamp.fromNSec(start_timestamp_.toNSec() + (ev_begin->t * 1000.00));
+                    // Get the current time
+                    event_buffer_current_time_.fromNSec(start_timestamp_.toNSec() + (ev_begin->t * 1000.00));
+
+                    /** In case the buffer is empty we set the starting time stamp **/
+                    if (event_buffer_.empty()) {
+                        // Get starting time
+                        event_buffer_start_time_ = event_buffer_current_time_;
+                    }
+
+                    /** Insert the events to the buffer **/
+                    auto inserter = std::back_inserter(event_buffer_);
+
+                    if (activity_filter_temporal_depth_ > 0) {
+                        /** When there is activity filter **/
+                        activity_filter_->process_output(ev_begin, ev_end, inserter);
+                    } else {
+                        /** When there is not activity filter **/
+                        std::copy(ev_begin, ev_end, inserter); 
+                    }
+
+                    /** Get the last time stamp **/
+                    event_buffer_current_time_.fromNSec(start_timestamp_.toNSec() + (ev_end-1)->t * 1000.00);
+                }
+
+                if ((event_buffer_current_time_ - event_buffer_start_time_) >= event_delta_t_) {
```

Comment:
why do you want to limit it ? why not publishing events to ROS as soon as you have them?

Reply:
I agree with Javier's implementation, this is useful for applications where you want to have buffers of events in a predefined time interval.
In any case, you can set the threshold to zero if you want to take all the events, so this modification is not blocking any implementation.

Reply:
I am not limiting. I am just buffering the n events that occurred between two consecutive frames. The reason is to visualize the gray-scale frame together with the corresponding CD events. I hope this picture helps. 
![atis_rendering](https://user-images.githubusercontent.com/4748753/71185023-f3ceeb00-227a-11ea-9c14-b8be70e09356.png)

This also helps to create afterwards tensors for PyTorch or TensorFlow.

Reply:
thanks, now it's clear!